src/libstd/macros.rs

```diff
@@ -24,6 +25,29 @@ macro_rules! panic {
     });
 }
 
+#[cfg(not(bootstrap))]
+#[doc(include = "../libcore/macros/panic.md")]
+#[macro_export]
+#[stable(feature = "rust1", since = "1.0.0")]
+#[allow_internal_unstable(libstd_sys_internals, track_caller)]
+macro_rules! panic {
+    () => ({
+        $crate::panic!("explicit panic")
+    });
+    ($msg:expr) => ({
+        $crate::rt::begin_panic($msg, $crate::panic::Location::caller())
```

Comment:
Why not make `begin_panic` `#[track_caller]`?

Reply:
I tried making all of these `#[track_caller]` on a different branch and had issues, but I think that may have been an issue with `begin_panic_fmt` and `rust_panic_with_hook`.

Reply:
I'll try it out.

Reply:
Looks like there's at least a dependency on the ABI in https://github.com/rust-lang/rust/blob/master/src/librustc_mir/interpret/intrinsics.rs#L397. Want me to go through and update this and others now, or you OK to wait until after we can get rid of the bootstrap cfg's?

Reply:
You need to get such a change in beta to be able to remove the cfg's later.